torch/distributed/algorithms/_comm_hooks/default_hooks.py

```diff
@@ -83,43 +83,83 @@ def allreduce_hook(state: DefaultState, grad: torch.Tensor):
         state (DefaultState): State information, configures pre- and post-division factors
         grad (torch.Tensor): A gradient for the local batch that needs to be communicated across ranks.
     """
+    # Average grad by pre-division factor. Together pre- and post-division factors
+    # lead to an overall averaging by world_size, required for consistency with PyTorch DDP.
+    # This is a two-step process to avoid potential underflow and overflow.
     if state.gradient_predivide_factor > 1:
         grad.div_(state.gradient_predivide_factor)
     dist.all_reduce(grad, group=state.process_group)
+    # Average grad by post-division factor. Together pre- and post-division factors
```

Comment:
It seems like the part of the comment starting with "Together..." is duplicated with above. This seems unnecessary.